cpp/src/rolling/rolling_collect_list.cu

```diff
@@ -0,0 +1,181 @@
+/*
```

Comment:
Note: This file is mostly utilities that were moved from `rolling_collect_list.cuh` to removed an anonymous namespace declared in the header file. The code here also contains the corrections per this PR: fix the memcheck error and correct the memory-resource usage.

Reply:
Thanks for clearing up the anonymous namespace. 

I'll review this PR more closely tomorrow, for the changes to `collect_list` utility functions. (It's a little hard to find the diffs, since the functions moved.) Am I right in understanding that the main change was to use the default memory resource for the temporary allocations?

Reply:
And also to use the memory resource for non-temporary allocations (e.g. the gather call).